src/main/java/org/threadly/examples/features/PrioritySchedulerExample.java

```diff
@@ -0,0 +1,146 @@
+package org.threadly.examples.features;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.ExecutionException;
+
+import org.threadly.concurrent.future.FutureUtils;
+import org.threadly.concurrent.future.ListenableFuture;
+import org.threadly.concurrent.PriorityScheduler;
+import org.threadly.concurrent.TaskPriority;
+
+
+/**
+ * Provides a basic example of the PriorityScheduler feature.
+ * 
+ * @author sreid8 - Sean Reid
+ *
+ */
+public final class PrioritySchedulerExample {
+  
+  /** static reference to the futures created */ 
+  private static List<ListenableFuture<?>> statFutures;
+  
+  /**
+   * Provided a PriorityScheduler that contains some basic ideas
+   * @return -> the PriorityScheduler with tasks added
+   */
+  public static PriorityScheduler getPriorityScheduler() {
+    final int numThreads = Runtime.getRuntime().availableProcessors() * 2;
+    final PriorityScheduler executor = new PriorityScheduler(numThreads);
+    List<ListenableFuture<?>> futures = new ArrayList<ListenableFuture<?>>(numThreads);
+    
+    futures.add(executor.submit(new Runnable() {
+      
+      @Override
+      public void run() {
+        /*
+         * This is the Starvable priority task that will be run.
+         * Starvable tasks run only when no other task needs to run
+         * It can be used to keep track of configuration:
+         * Example use case: closing sockets that no longer have hosts
+         * available on the other side.
+         */
+        
+      }
+    },
+    /*The priority of the task we are submitting*/TaskPriority.Starvable));
+    
+    futures.add(executor.submit(new Runnable() {
+      
+      @Override
+      public void run() {
+        /*
+         * This is the Low priority task that will be run.
+         * Low priority tasks are run before Starvable tasks,
+         * but exactly when it will be run is enforced by the 
+         * PrioritySchedulerService.
+         * Example use case: cache cleaning or other cleanup work
```

Comment:
Ah, sorry, I was a bit confusing there.

I was thinking you would implement a "dispose" or "shutdown" that would just remove the recurring runnable...but I can see how that would be a little awkward since you are constructing the priority scheduler (it would make more sense if it was passed in).  The idea being "I constructed this, it ran for a while, and now I am no longer interested in it, so stop it, and let me use my resources for something else".

No matter the case it would not send an interrupt to the running task.  It would just prevent future executions.

Needless to say this is example code, so I don't know how much sense it needs to make or not.  I am fine if you want to omit it.